### PR TITLE
Add updated gnome packages, and unstackguarded packages

### PR DESCRIPTION
**Pull request by satmandu**
- Adds required deps for gnome_tweaks. (graphical not for i686)
- removes stackguard from a bunch of packages.
- updates glib

Works properly:
- [x] x86_64

Builds properly:
- [x] x86_64
- [x] i686 (non-graphical compiles. mutter will NOT be supported on the older kernel.)
- [x] armv7l

### CODE REVIEW

File: packages/gnome_session.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# https://github.com/archlinux/svntogit-packages/raw/packages/gnome-session/trunk/PKGBUILD
+
+require 'package'
+
+class Gnome_session < Package
+  description 'The GNOME Session Handler'
+  homepage 'https://gitlab.gnome.org/GNOME/gnome-session'
+  @_ver = '40.beta'
+  version @_ver
+  compatibility 'x86_64 , armv7l'
```
Comment (uberhacker) [inline packages/gnome_session.rb:11]: Missing aarch64?

File: packages/gnome_shell.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Adapted from Arch Linux gnome-shell PKGBUILD at:
+# https://github.com/archlinux/svntogit-packages/raw/packages/gnome-shell/trunk/PKGBUILD
+
+require 'package'
+
+class Gnome_shell < Package
+  description 'Next generation desktop shell'
+  homepage 'https://wiki.gnome.org/Projects/GnomeShell'
+  version '40.beta'
+  compatibility 'x86_64 , armv7l'
```
Comment (uberhacker) [inline packages/gnome_shell.rb:10]: Missing aarch64?

File: packages/gnome_shell.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  source_url 'https://github.com/GNOME/gnome-shell/archive/40.beta.tar.gz'
```
Comment (uberhacker) [inline packages/gnome_shell.rb:11]: `source_url "https://github.com/GNOME/gnome-shell/archive/#{version}.tar.gz"`?

File: packages/gnome_tweaks.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# Adapted from Arch Linux gnome-tweaks PKGBUILD at:
+# https://github.com/archlinux/svntogit-packages/raw/packages/gnome-tweaks/trunk/PKGBUILD
+
+require 'package'
+
+class Gnome_tweaks < Package
+  description 'Graphical interface for advanced GNOME 3 settings Tweak Tool'
+  @_ver = '40.beta'
+  version @_ver
+  compatibility 'x86_64 , armv7l'
```
Comment (uberhacker) [inline packages/gnome_tweaks.rb:10]: Missing aarch64?

File: packages/gnome_tweaks.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  source_url "https://gitlab.gnome.org/GNOME/gnome-tweaks/-/archive/40.beta/gnome-tweaks-#{@_ver}.tar.bz2"
```
Comment (uberhacker) [inline packages/gnome_tweaks.rb:11]: `source_url "https://gitlab.gnome.org/GNOME/gnome-tweaks/-/archive/#{@_ver}/gnome-tweaks-#{@_ver}.tar.bz2"`?

File: packages/mutter.rb [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+# https://github.com/archlinux/svntogit-packages/raw/packages/mutter/trunk/PKGBUILD
+
+require 'package'
+
+class Mutter < Package
+  description 'A window manager for GNOME'
+  homepage 'https://gitlab.gnome.org/GNOME/mutter'
+  version '40.beta'
+  compatibility 'x86_64 , armv7l'
+  source_url 'https://download.gnome.org/core/40/40.beta/sources/mutter-40.beta.tar.xz'
```
Comment (uberhacker) [inline packages/mutter.rb:11]: `source_url "https://download.gnome.org/core/40/#{version}/sources/mutter-#{version}.tar.xz"`?

File: packages/mutter.rb [hunk 7]
```diff
@@ -0,0 +1,120 @@
+# Adapted from Arch Linux mutter PKGBUILD at:
+# https://github.com/archlinux/svntogit-packages/raw/packages/mutter/trunk/PKGBUILD
+
+require 'package'
+
+class Mutter < Package
+  description 'A window manager for GNOME'
+  homepage 'https://gitlab.gnome.org/GNOME/mutter'
+  version '40.beta'
+  compatibility 'x86_64 , armv7l'
```
Comment (uberhacker) [inline packages/mutter.rb:10]: Missing aarch64?

File: packages/mutter.rb [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  depends_on 'libwacom'
+  depends_on 'startup_notification'
+  depends_on 'pipewire'
+  depends_on 'libcanberra'
+  depends_on 'ccache' => :build
+
+  case ARCH
+  when 'i686'
+
+  end
```
Comment (uberhacker) [inline packages/mutter.rb:40]: What is happening here? :)

File: packages/sane_backends.rb [hunk 9]
```diff
@@ -3,33 +3,34 @@
 class Sane_backends < Package
   description 'Scanner Access Now Easy - Backends'
   homepage 'http://www.sane-project.org/'
-  version '1.0.27'
+  version '1.0.32'
   compatibility 'all'
-  source_url 'https://gitlab.com/sane-project/backends/-/archive/RELEASE_1_0_27/backends-RELEASE_1_0_27.tar.bz2'
-  source_sha256 '6a4034a9d29255bb40cf5a19f3d85dd7ba4fe6452eb9b1ae4385bc8175493ec4'
+  source_url 'https://gitlab.com/sane-project/backends/uploads/104f09c07d35519cc8e72e604f11643f/sane-backends-1.0.32.tar.gz'
```
Comment (uberhacker) [inline packages/sane_backends.rb:8]: `source_url "https://gitlab.com/sane-project/backends/uploads/104f09c07d35519cc8e72e604f11643f/sane-backends-#{version}.tar.gz"`?

File: packages/gnome_session.rb (hunk 1 above)
Comment (satmandu) [reply packages/gnome_session.rb:11]: Ah yes, I need to modify my script to add those...

File: packages/mutter.rb (hunk 6 above)
Comment (satmandu) [reply packages/mutter.rb:11]: It's a beta URL, they might change the final release URL config, so it doesn't make sense to overautomate this...

File: packages/mutter.rb (hunk 8 above)
Comment (satmandu) [reply packages/mutter.rb:40]: i686 is not supported on mutter... so I gave up on the patching... I'll get rid of that.

File: packages/sane_backends.rb (hunk 9 above)
Comment (satmandu) [reply packages/sane_backends.rb:8]: See the hash in the URL? That doesn't seem like a stable scriptable URL.

File: packages/gnome_tweaks.rb (hunk 5 above)
Comment (satmandu) [reply packages/gnome_tweaks.rb:11]: Sure, but this might move to a 40/40.something at some point...

File: packages/sane_backends.rb (hunk 9 above)
Comment (uberhacker) [reply packages/sane_backends.rb:8]: How about `source_url "https://gitlab.com/sane-project/backends/-/archive/#{version}/backends-#{version}.tar.bz2"`?

File: packages/gnome_session.rb [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+# https://github.com/archlinux/svntogit-packages/raw/packages/gnome-session/trunk/PKGBUILD
+
+require 'package'
+
+class Gnome_session < Package
+  description 'The GNOME Session Handler'
+  homepage 'https://gitlab.gnome.org/GNOME/gnome-session'
+  @_ver = '40.beta'
+  version @_ver
+  compatibility 'x86_64 , aarch64, armv7l'
```
Comment (uberhacker) [inline packages/gnome_session.rb:11]: Can we clean up the spacing?  I don't think you even need commas.

File: packages/sane_backends.rb (hunk 9 above)
Comment (satmandu) [reply packages/sane_backends.rb:8]: That works!

File: packages/sane_backends.rb [hunk 11]
```diff
@@ -5,7 +5,7 @@ class Sane_backends < Package
   homepage 'http://www.sane-project.org/'
   version '1.0.32'
   compatibility 'all'
-  source_url 'https://gitlab.com/sane-project/backends/uploads/104f09c07d35519cc8e72e604f11643f/sane-backends-1.0.32.tar.gz'
+  source_url "https://gitlab.com/sane-project/backends/-/archive/#{version}/backends-#{version}.tar.bz2"
   source_sha256 '3a28c237c0a72767086202379f6dc92dbb63ec08dfbab22312cba80e238bb114'
```
Comment (uberhacker) [inline packages/sane_backends.rb:9]: sha256 is the same?

Comment (satmandu) [reply packages/sane_backends.rb:9]: Most definitely not.

File: packages/libssh.rb [hunk 12]
```diff
@@ -3,41 +3,46 @@
 class Libssh < Package
   description 'libssh is a multiplatform C library implementing the SSHv2 and SSHv1 protocol on client and server side.'
   homepage 'https://www.libssh.org/'
-  version '0.9.5'
+  @_ver = '0.9.5'
+  version "#{@_ver}-1"
   compatibility 'all'
-  source_url 'https://www.libssh.org/files/0.9/libssh-0.9.5.tar.xz'
+  @_ver_prelastdot = @_ver.rpartition('.')[0]
```
Comment (supechicken) [inline packages/libssh.rb:9]: I like this method :)

File: packages/sane_backends.rb [hunk 13]
```diff
@@ -3,33 +3,34 @@
 class Sane_backends < Package
   description 'Scanner Access Now Easy - Backends'
   homepage 'http://www.sane-project.org/'
-  version '1.0.27'
+  version '1.0.32'
   compatibility 'all'
-  source_url 'https://gitlab.com/sane-project/backends/-/archive/RELEASE_1_0_27/backends-RELEASE_1_0_27.tar.bz2'
-  source_sha256 '6a4034a9d29255bb40cf5a19f3d85dd7ba4fe6452eb9b1ae4385bc8175493ec4'
+  source_url "https://gitlab.com/sane-project/backends/-/archive/#{version}/backends-#{version}.tar.bz2"
+  source_sha256 'c0425aad44e4f79dbf2f123d3060d690c5c328ef58316e0ad0322c61d63a4783'
```
Comment (uberhacker) [inline packages/sane_backends.rb:9]: Did you generate new binaries with this new release?

Comment (satmandu) [reply packages/sane_backends.rb:9]: Not once again once I changed the URL a second time. I can do so if needed...

Comment (satmandu) [reply packages/sane_backends.rb:9]: Only the old archive works to build. not the new one. So reverting the URL change.

**Pull request merged**